src/components/radio-button/radio-button.component.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useCallback } from "react";
+import { MarginProps } from "styled-system";
+import invariant from "invariant";
+import { CommonCheckableInputProps } from "../../__internal__/checkable-input";
+import RadioButtonStyle from "./radio-button.style";
+import CheckableInput from "../../__internal__/checkable-input/checkable-input.component";
+import RadioButtonSvg from "./radio-button-svg.component";
+import { filterStyledSystemMarginProps } from "../../style/utils";
+import { TooltipProvider } from "../../__internal__/tooltip-provider";
+
+interface InternalRadioButtonProps {
+  inline?: boolean;
+  required?: boolean;
+}
+
+export interface RadioButtonProps
+  extends CommonCheckableInputProps,
+    MarginProps {
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-component"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-element"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-role"?: string;
+  /** When true, sets the component in line (for RadioButtonGroup) */
+  inline?: boolean;
+  /** Text alignment of the label */
+  labelAlign?: "left" | "right";
+  /** Accepts a callback function which is triggered on click event */
+  onClick?: (ev: React.MouseEvent<HTMLInputElement>) => void;
+  /** the value of the Radio Button, passed on form submit */
+  value: string;
+  /** Overrides the default tooltip position */
+  tooltipPosition?: "top" | "bottom" | "left" | "right";
+  /** Aria label for rendered help component */
+  helpAriaLabel?: string;
+}
+
+const RadioButton = React.forwardRef<
+  HTMLInputElement,
+  RadioButtonProps & InternalRadioButtonProps
+>(
+  (
+    {
+      autoFocus,
+      checked,
+      disabled,
+      fieldHelp,
+      fieldHelpInline,
+      id,
+      inline,
+      inputWidth,
+      label,
+      labelHelp,
+      labelAlign,
+      labelSpacing = 1,
+      labelWidth,
+      name,
+      onChange,
+      onBlur,
+      onFocus,
+      value,
+      reverse = false,
+      required,
+      size,
+      error,
+      warning,
+      info,
+      tooltipPosition,
+      "data-component": dataComponent = "radio-button",
+      "data-element": dataElement,
+      "data-role": dataRole,
+      helpAriaLabel,
+      ...props
+    }: RadioButtonProps & InternalRadioButtonProps,
+    ref
+  ) => {
+    const marginProps = filterStyledSystemMarginProps(props);
+    const handleChange = useCallback(
+      (event: React.ChangeEvent<HTMLInputElement>) => {
+        /* istanbul ignore else */
+        if (onChange) {
+          onChange(event);
```

Comment:
FYI: You could always do `onChange?.(event)`

Reply:
good point - I keep forgetting about optional chaining